### PR TITLE
[Merged by Bors] - feat: Cardinality of a finset under a surjective map

### PR DESCRIPTION
**Pull request by YaelDillies**
Prove a few easy (but annoying to prove on the fly) results about cardinality of finsets:
* If `f` is surjective from `s` to `t`, then `t.card ≤ s.card`.
* If two finsets `t`, `u` are subsets of `s` and `s.card < t.card + u.card`, then `t` and `u` are not disjoint.

Rename `Finset.card_le_card_of_inj_on` to `Finset.card_le_card_of_injOn` and state it using `Set.InjOn`. Golf `Finset.surj_on_of_inj_on_of_card_le`.

From LeanAPAP

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/Finset/Card.lean [hunk 1]
```diff
@@ -441,57 +441,51 @@ See `Fintype.card_bijective` for the version where `s` and `t` are `univ`. -/
 lemma card_bijective (e : α → β) (he : e.Bijective) (hst : ∀ i, i ∈ s ↔ e i ∈ t) :
     s.card = t.card := card_equiv (.ofBijective e he) hst
 
-end bij
-
-theorem card_le_card_of_inj_on {t : Finset β} (f : α → β) (hf : ∀ a ∈ s, f a ∈ t)
-    (f_inj : ∀ a₁ ∈ s, ∀ a₂ ∈ s, f a₁ = f a₂ → a₁ = a₂) : s.card ≤ t.card := by
+lemma card_le_card_of_injOn (f : α → β) (hf : ∀ a ∈ s, f a ∈ t) (f_inj : (s : Set α).InjOn f) :
```
Comment (Ruben-VandeVelde) [inline Mathlib/Data/Finset/Card.lean:444]: Please add the deprecated alias

Comment (YaelDillies) [reply Mathlib/Data/Finset/Card.lean:444]: Uh weird, pretty sure I added it already but I can't find it

File: Mathlib/Data/Finset/Card.lean [hunk 2]
```diff
@@ -441,57 +441,51 @@ See `Fintype.card_bijective` for the version where `s` and `t` are `univ`. -/
...
+    s.card ≤ t.card := by
   classical
   calc
     s.card = (s.image f).card := (card_image_of_injOn f_inj).symm
     _      ≤ t.card           := card_le_card <| image_subset_iff.2 hf
-#align finset.card_le_card_of_inj_on Finset.card_le_card_of_inj_on
+#align finset.card_le_card_of_inj_on Finset.card_le_card_of_injOn
```
Comment (YaelDillies) [inline Mathlib/Data/Finset/Card.lean:450]:
```suggestion
@[deprecated (since := "2024-06-01")] alias card_le_card_of_inj_on := card_le_card_of_injOn
```

**Review by Ruben-VandeVelde — COMMENTED** [review]
lgtm, please put on the queue when green

**Review by Ruben-VandeVelde — APPROVED** [review]

**Comment by YaelDillies** [comment]
maintainer merge

**Review by jcommelin — COMMENTED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**